packages/core/src/types/types.ts

```diff
@@ -63,3 +64,56 @@ export interface CloudWatchDataTracker {
 	logEvents: InputLogEvent[];
 	verifiedLogGroup?: LogGroup;
 }
+
+// TODO: the following types are TBD this is only an initial draft
+export interface JWT {
+	payload: JSON;
+	header: JSON;
+	signature: string;
+	toString: () => string;
+	isValid: () => boolean;
+}
+
+export type JWTS = {
+	accessToken?: JWT;
+	idToken?: JWT;
+};
+
+export type AmplifyUserSession = {
+	isLoggedIn: boolean;
+	username?: string;
+	credentials?: ICredentials;
+	jwts?: JWTS;
+};
+
+export type GetUserSessionOptions = {
+	refresh?: boolean;
+};
+
+export interface AmplifyUserSessionProvider {
+	getUserSession: (
+		options?: GetUserSessionOptions
+	) => Promise<AmplifyUserSession>;
+	listenUserSession(): Observable<AmplifyUserSession>;
+}
+
+export type FrontendConfig = {
+	Auth: {
+		sessionProvider?: AmplifyUserSessionProvider;
+	} | null;
+};
+
+export type ResourceConfig = {
+	Auth?: {
+		userPoolId?: string;
+		identityPoolId?: string;
+		userPoolWebClientId?: string;
+	};
+	API?: {};
+	Analytics?: {};
+	Storage?: {};
+	DataStore?: {};
+	Predictions?: {};
+	Interactions?: {};
+	Notifications?: {};
```

Comment:
Mega-nit: can we make this alphabetical?

```suggestion
	API?: {};
	Analytics?: {};
	Auth?: {
		userPoolId?: string;
		identityPoolId?: string;
		userPoolWebClientId?: string;
	};
	DataStore?: {};
	Interactions?: {};
	Notifications?: {};
	Predictions?: {};
	Storage?: {};
```